### DISCUSSION TITLE
Issue trying to mock an overloaded method

### DISCUSSION THREAD

**Discussion by thitemple** [Typescript-TDD/ts-auto-mock#1015] [Q&A]
Hi I'm trying to create a mock for express' Request type and I'm getting an error when defining a value for its `get` method as defined here

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/bd4c4ba6aa8c6ad9739c17d1023349a5f9f496cd/types/express-serve-static-core/index.d.ts#L396

The method has the following overload:

```ts
get(name: 'set-cookie'): string[] | undefined;
get(name: string): string | undefined;
```

When I write the following

```ts
createMock<Request>({
      get(name: string) {
        if (!header) return undefined;
        if (name.toLowerCase() === header[0].toLowerCase()) return header[1];
        return undefined;
      },
    });
```

I get this error:

```
Type '(name: string) => string | undefined' is not assignable to type '{ (name: string): string | undefined; (name: "set-cookie"): string[] | undefined; }'.
  Type 'string | undefined' is not assignable to type 'string[] | undefined'.
    Type 'string' is not assignable to type 'string[] | undefined'.ts(2322)
```

Is there a solution for that?

**Comment by uittorio** [comment]
Hey, its seems like a problem with typescript! I havent tried yet but your function needs to respect the interface provided!
